planner/core/expression_rewriter.go

```diff
@@ -941,10 +941,26 @@ func (er *expressionRewriter) isNullToExpression(v *ast.IsNullExpr) {
 }
 
 func (er *expressionRewriter) positionToScalarFunc(v *ast.PositionExpr) {
-	if v.N > 0 && v.N <= er.schema.Len() {
-		er.ctxStack = append(er.ctxStack, er.schema.Columns[v.N-1])
+	pos := v.N
+	str := strconv.Itoa(pos)
+	stkLen := len(er.ctxStack)
+	if v.P != nil {
+		val := er.ctxStack[stkLen-1]
+		intNum, isNull, err := expression.GetIntFromConstant(er.ctx, val)
```

Comment:
how about to handle `err` immediately after this function call?
```go
if err != nil {
	// handle err
}
er.ctxStack = er.ctxStack[:stkLen-1]
if isNull {
	// handle NULL position
}
// update pos, str
```

Reply:
If we change the code as you mentioned, the error message setting code (i.e., ``er.err = ErrUnknownColumn.GenWithStackByArgs(str, clauseMsg[er.b.curClause])``) will be duplicated in the both places: (1) the return error checking (i.e., ``err != nil``) and (2) the range error checking of ``pos``.

Reply:
> the return error checking
do you mean this line: https://github.com/pingcap/tidb/pull/8206/files#diff-33ebff1614e7db77b69dd02c2eee26c2R960

Reply:
yes.